ui/src/routes/devices.$id.settings.macros.tsx

```diff
@@ -0,0 +1,255 @@
+                <span className="flex flex-col items-start gap-1">
+                  {macro.steps.map((step, stepIndex) => {
+                    const StepIcon = stepIndex === 0 ? LuMoveRight : LuCornerDownRight;
+
+                    return (
+                      <span key={stepIndex} className="inline-flex items-center">
+                        <StepIcon className="mr-1 text-slate-400 dark:text-slate-500 h-3 w-3 flex-shrink-0" />
+                        <span className="bg-slate-50 dark:bg-slate-800 px-2 py-0.5 rounded-md border border-slate-200/50 dark:border-slate-700/50">
+                          {(Array.isArray(step.modifiers) && step.modifiers.length > 0) || (Array.isArray(step.keys) && step.keys.length > 0) ? (
+                            <>
+                              {Array.isArray(step.modifiers) && step.modifiers.map((modifier, idx) => (
+                                <Fragment key={`mod-${idx}`}>
+                                  <span className="font-medium text-slate-600 dark:text-slate-200">
+                                    {modifierDisplayMap[modifier] || modifier}
+                                  </span>
+                                  {idx < step.modifiers.length - 1 && (
+                                    <span className="text-slate-400 dark:text-slate-600"> + </span>
+                                  )}
+                                </Fragment>
+                              ))}
+
+                              {Array.isArray(step.modifiers) && step.modifiers.length > 0 && Array.isArray(step.keys) && step.keys.length > 0 && (
+                                <span className="text-slate-400 dark:text-slate-600"> + </span>
+                              )}
+
+                              {Array.isArray(step.keys) && step.keys.map((key, idx) => (
+                                <Fragment key={`key-${idx}`}>
+                                  <span className="font-medium text-blue-600 dark:text-blue-400">
+                                    {keyDisplayMap[key] || key}
+                                  </span>
+                                  {idx < step.keys.length - 1 && (
+                                    <span className="text-slate-400 dark:text-slate-600"> + </span>
+                                  )}
+                                </Fragment>
+                              ))}
+                            </>
+                          ) : (
+                            <span className="font-medium text-slate-500 dark:text-slate-400">Delay only</span>
+                          )}
+                          <span className="ml-1 text-slate-400 dark:text-slate-500">({step.delay}ms)</span>
+                        </span>
+                      </span>
+                    );
+                  })}
+                </span>
+              </p>
+            </div>
+
+            <div className="flex items-center gap-1 ml-4">
+              <Button
```

Comment:
What do you say about adding a button with 
```
className="text-red-500"
theme="light"
LeadingIcon={LuTrash2}
```

dont forget to confirm it, though

Reply:
added